### PR TITLE
feat: add bootstrap step

### PR DESCRIPTION
**Pull request by rjerrems**
Adding the first part of the example foundation. I will submit each section as a separate PR to make it more digestible to review and document each individually.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.6.0
+DOCKER_IMAGE_DEVELOPER_TOOLS := cft/developer-tools
+REGISTRY_URL := gcr.io/cloud-foundation-cicd
+
+# Execute lint tests within the docker container
+.PHONY: docker_test_lint
+docker_test_lint:
+	docker run --rm -it \
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
```
Comment (ericyz) [inline Makefile]: How could I find what this image is about?

File: 0-bootstrap/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+1. A GCP [Billing Account](https://cloud.google.com/billing/docs/how-to/manage-billing-account)
+1. Cloud Identity / Gsuite groups for organization and billing admins
+1. Membership in the `group_org_admins` group for user running terraform
+
+Further details of permissions required and resources created, can be found in the bootstrap module [documentation.](https://github.com/terraform-google-modules/terraform-google-bootstrap)
+
+## Usage
+
+1. Change into 0-bootstrap folder
+1. Rename terraform.example.tfvars to terraform.tfvars and update the file with values from your environment.
```
Comment (morgante) [inline 0-bootstrap/README.md]: We should include explicit instructions (ex. `cp terraform.example.tfvars terraform.tfvars`.

File: 0-bootstrap/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+1. Run `terraform init`
+1. Run `terraform plan` and review output
+1. Run `terraform apply`
+1. Upload terraform state to created GCS bucket `gsutil cp terraform.tfstate gs://STATE_BUCKET/terraform/bootstrap/state/default.tfstate`
```
Comment (morgante) [inline 0-bootstrap/README.md]: No need to do this manually. Once you reconfigure backend, Terraform prompts you to copy the state.

File: 0-bootstrap/backend.tf.example [hunk 4]
```diff
@@ -0,0 +1,6 @@
+terraform {
+  backend "gcs" {
+    bucket = "STATE_BUCKET"
+    prefix = "terraform/bootstrap/state"
+  }
+}
```
Comment (morgante) [inline 0-bootstrap/backend.tf.example]: Tiny nit: files should have a newline at the end.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Please note that this file was generated from [terraform-google-module-template](https://github.com/terraform-google-modules/terraform-google-module-template).
+# Please make sure to contribute relevant changes upstream!
+
+# Make will use bash instead of sh
+SHELL := /usr/bin/env bash
+
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.6.0
```
Comment (morgante) [inline Makefile]:
```suggestion
DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0
```

File: 0-bootstrap/README.md (hunk 2 above)
Comment (rjerrems) [reply 0-bootstrap/README.md]: Make sense, fixed in next push.

File: 0-bootstrap/README.md (hunk 3 above)
Comment (rjerrems) [reply 0-bootstrap/README.md]: This is cool, did not know about it. Have updated docs in next commit.

**Comment by rjerrems** [comment]
Thanks for the review @morgante - I have fixed according to your feedback and added a PR for linting:

https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/609

**Pull request merged**